scripts/build/wrap_binary_in_venv.py

```diff
@@ -0,0 +1,118 @@
+from logging import getLogger, Formatter, StreamHandler, INFO, DEBUG
+from os import stat, chmod
+from os.path import abspath
+from stat import S_IXUSR
+from string import Template
+
+LOG = getLogger('VirtualenvWrapper')
+
+msg_formatter = Formatter('[%(levelname)s] - %(message)s')
+log_handler = StreamHandler()
+log_handler.setFormatter(msg_formatter)
+LOG.setLevel(INFO)
+LOG.addHandler(log_handler)
+
+
+def add_executable_permission(path):
+    current_stat = stat(path)
+    chmod(path, current_stat.st_mode | S_IXUSR)
+
+
+def generate_content(binary_path, output_path, virtual_environment_path):
+
+    template_content = '\n'.join([
+        r'#!/bin/bash',
+        r'source "${virtual_environment_path}/bin/activate"',
+        r'exec "${binary_path}" "$$@"'])
+
+    template_vars = {
+        'binary_path': binary_path,
+        'output_path': output_path,
+        'virtual_environment_path': virtual_environment_path}
+
+    LOG.debug('Using template:\n%s\nwith the following substitution:\n%s',
+              template_content, template_vars)
+
+    return Template(template_content).substitute(template_vars)
+
+
+def create_venv_wrapper(binary, output, environment):
+    """
+    Create a runnable file which wraps another runnable file. In addition to
+    forwarding all arguments, the wrapper also activates the virtual
+    environment before running the wrapped runnable file.
+    """
+
+    binary_path = abspath(binary)
+
+    LOG.debug('Absolut binary path: %s', binary_path)
+
+    add_executable_permission(binary_path)
+
+    output_path = abspath(output)
+    virtual_environment_path = abspath(environment)
+
+    wrapper_content = generate_content(
+        binary_path, output_path, virtual_environment_path)
+
+    with open(output_path, 'w') as wrapper_file:
+        wrapper_file.write(wrapper_content)
+
+    add_executable_permission(output_path)
+
+
+if __name__ == "__main__":
+    description = '''Wrap python binaries in virtualenv. This utility can be
+        used to create standalone runnable python binaries, which no longer
+        require the user to explicitly activate the projects virtual
+        environment.'''
+
+    parser = ArgumentParser(
+        formatter_class=RawDescriptionHelpFormatter,
+        description=description)
+
+    parser.add_argument('-b', '--binary',
+                        required=True,
+                        action='store',
+                        dest='binary',
+                        help="Path to the runnable file to wrap.")
+
+    parser.add_argument('-o', '--output',
+                        required=True,
+                        action='store',
+                        dest='output',
+                        help="Path to the created wrapper file.")
+
+    parser.add_argument('-e', '--environment',
+                        required=True,
+                        action='store',
+                        dest='environment',
+                        help=(
```

Comment:
Why do we need this extra parenthesis?

```suggestion
                        help="Path to the root of the virtual environment, that "
                             "is used as a wrapper."
```

Reply:
Fixed.